sdk/core/azure-core/src/main/java/com/azure/core/implementation/http/policy/InstrumentationPolicy.java

```diff
@@ -122,16 +137,16 @@ private void addPostSamplingAttributes(Context span, HttpRequest request) {
     private void endSpan(HttpResponse response, Context span) {
         if (response != null) {
             int statusCode = response.getStatusCode();
-            tracer.setAttribute(HTTP_STATUS_CODE, statusCode, span);
+            tracer.setAttribute(HTTP_RESPONSE_STATUS_CODE, statusCode, span);
             String requestId = response.getHeaderValue(SERVICE_REQUEST_ID_HEADER);
             if (requestId != null) {
                 tracer.setAttribute(SERVICE_REQUEST_ID_ATTRIBUTE, requestId, span);
             }
 
-            tracer.end((statusCode >= 400) ? "error" : null, null, span);
+            tracer.end((statusCode >= 400) ? String.valueOf(statusCode) : null, null, span);
```

Comment:
I didn't follow if it's intentional to fall through and call tracer.end() again below?
```suggestion
            tracer.end((statusCode >= 400) ? String.valueOf(statusCode) : null, null, span);
            return;
```

Reply:
great catch, thanks!